handler.h

```diff
@@ -0,0 +1,63 @@
+#ifndef HANDLER_H_
+#define HANDLER_H_
+
+#include <aws/lambda-runtime/runtime.h>
+#include <aws/core/Aws.h>
+#include <aws/core/utils/logging/LogLevel.h>
+#include <aws/core/utils/logging/ConsoleLogSystem.h>
+#include <aws/core/utils/logging/LogMacros.h>
+#include <aws/core/utils/json/JsonSerializer.h>
+#include <aws/core/utils/HashingUtils.h>
+#include <aws/core/platform/Environment.h>
+#include <aws/core/client/ClientConfiguration.h>
+#include <aws/core/auth/AWSCredentialsProvider.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/GetObjectRequest.h>
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <sstream>
+#include <string>
+
+#include "s3_audio_processing.h"
+#include "transcription.h"
+#include "common.h"
+
+using namespace aws::lambda_runtime;
+
+char const TAG[] = "WHISPER_HANDLER";
+
+static invocation_response whisper_handler(invocation_request const &req, Aws::S3::S3Client const &client) {
+  using namespace Aws::Utils::Json;
+  AWS_LOGSTREAM_INFO(TAG, "Invoked with payload: " << req.payload);
+
+  JsonValue json(req.payload);
+  if (!json.WasParseSuccessful()) {
+      return invocation_response::failure("Failed to parse input JSON", "InvalidJSON");
+  }
+
+  auto v = json.View();
+
+  if (!v.ValueExists("s3bucket") || !v.ValueExists("s3key") || !v.GetObject("s3bucket").IsString() ||
+      !v.GetObject("s3key").IsString()) {
+      return invocation_response::failure("Missing input value s3bucket or s3key", "InvalidJSON");
+  }
```

Comment:
I wonder if there's something like [`serde`](https://serde.rs/) for cpp so that you don't have to do all this manual checking for deserializing the fields...

Reply:
Actually, it seems like there's something: https://github.com/serde-cpp/serde-cpp

Reply:
But it doesn't seem to support JSON yet... nevermind...